reframe/frontend/autodetect.py

```diff
@@ -201,12 +200,18 @@ def detect_topology():
         if not found_procinfo:
             # No topology found, try to auto-detect it
             getlogger().debug(f'> no topology file found; auto-detecting...')
+            temp_modules = rt.system.preload_environ.modules
             if _is_part_local(part):
+                temp_modules += part.local_env.modules
                 # Unconditionally detect the system for fully local partitions
-                part.processor._info = cpuinfo()
+                with runtime.temp_environment(modules=temp_modules):
```

Comment:
You should also pass any `variables` that are set here. Perhaps we could modify the `temp_environment` to recognise an environment argument as well.

Reply:
I have added the variables but I am not sure it's worth it to add the environment in `temp_environment`. In case we do add it, should we call it with either environment or modules+vars? Or should we be able to call it with all three and make a new environment based on those?

Reply:
Let's leave it as of now. That's fine.